src/main/java/edu/wpi/first/shuffleboard/json/WidgetSaver.java

```diff
@@ -0,0 +1,42 @@
+package edu.wpi.first.shuffleboard.json;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import edu.wpi.first.shuffleboard.sources.Sources;
+import edu.wpi.first.shuffleboard.widget.Widget;
+import edu.wpi.first.shuffleboard.widget.Widgets;
+import javafx.beans.property.Property;
+
+@AnnotatedTypeAdapter(forType = Widget.class)
+public class WidgetSaver implements ElementTypeAdapter<Widget> {
+  @Override
+  public JsonElement serialize(Widget src, JsonSerializationContext context) {
+    JsonObject object = new JsonObject();
+    object.addProperty("_type", src.getName());
```

Comment:
The extra properties could go in a nested map. Or we just blacklist "type" and "source" as property names